src/uikit.cs

```diff
@@ -331,7 +331,13 @@ interface NSAttributedStringAttachmentConveniences {
 	[DisableDefaultCtor]
 	[Abstract] // abstract class that should not be used directly
 	[BaseType (typeof (NSObject))]
-	interface UIFeedbackGenerator {
+	interface UIFeedbackGenerator : UIInteraction {
+		// intancetype -> UIFeedbackGenerator, inlined as subclasses should return the same type as the instance
```

Comment:
```suggestion
		// instancetype -> UIFeedbackGenerator, inlined as subclasses should return the same type as the instance
```

Reply:
I don't think you can do it this way, you have to duplicate the binding in each subclass.

This method will always create a `UIFeedbackGenerator`, even when called from a subclass, and the fact that we cast the return value to the expected managed type doesn't actually make the returned instance of the corresponding native type.

Reply:
Great catch, this should be fixed now